src/e2e-node/e2e.test.ts

```diff
@@ -540,6 +542,54 @@ describe.each(serversUnderTest)(
       await deleteSolidDataset(datasetWithoutAclUrl, { fetch: session.fetch });
     });
 
+    it("can update Things containing Blank Nodes in different instances of the same SolidDataset", async () => {
+      const session = await getSession();
+      const regularPredicate = "https://arbitrary.vocab/regular-predicate";
+      const blankNodePredicate = "https://arbitrary.vocab/blank-node-predicate";
+
+      // Prepare the Resource on the Pod
+      let newThing = createThing({ name: "e2e-test-thing-with-blank-node" });
+      newThing = setBoolean(newThing, regularPredicate, true);
+      newThing = setTerm(newThing, blankNodePredicate, blankNode());
+      let newDataset = createSolidDataset();
+      newDataset = setThing(newDataset, newThing);
+
+      const datasetUrl = `${rootContainer}solid-client-tests/node/blank-node-updates-${session.info.sessionId}.ttl`;
+      await saveSolidDatasetAt(datasetUrl, newDataset, {
+        fetch: session.fetch,
+      });
+
+      // Fetch the initialised SolidDataset for the first time,
+      // and change the non-blank node value:
+      const initialisedDataset = await getSolidDataset(datasetUrl, {
+        fetch: session.fetch,
+      });
+      const initialisedThing = getThing(
+        initialisedDataset,
+        datasetUrl + "#e2e-test-thing-with-blank-node"
+      )!;
+
+      const updatedThing = setBoolean(
+        initialisedThing,
+        regularPredicate,
+        false
+      );
+
+      // Now fetch the Resource again, and try to insert the updated Thing into it:
+      const refetchedDataset = await getSolidDataset(datasetUrl, {
+        fetch: session.fetch,
+      });
+      const updatedDataset = setThing(refetchedDataset, updatedThing);
+      await expect(
+        saveSolidDatasetAt(datasetUrl, updatedDataset, {
+          fetch: session.fetch,
+        })
+      ).resolves.not.toThrow();
+
+      // Clean up after ourselves
+      await deleteSolidDataset(datasetUrl, { fetch: session.fetch });
```

Comment:
In general, shouldn't this clean-up be in a `finally` block (in case the test fails after creating this resource on the remote Pod)? But even that still wouldn't guarantee we **_always_** clean up properly (as the connection could drop before the DELETE arrives), so I'd always create a fixed-named resource (specifically named just for this test), and just attempt to delete that resource (whether it exists or not) as the first step of the test. (I assume this is why you use `${session.info.sessionId}` in the resource name - but I don't think that's necessary really.)

Reply:
Ah yes, good point, I'll change that. I specifically avoided the fixed-named resource for two reasons:

- Multiple end-to-end tests may be running at the same time (e.g. in CI, in different versions of Node or different operating systems), so I don't want those to read and write to the same resource.
- I'd prefer not to risk have a failing test influence future tests, or a failing cleanup at the start of the test to influence its behaviour.

Given that an empty, unused resource is cheap, that felt like a fair trade-off.